cmd/guacone/cmd/vulnerability.go

```diff
@@ -561,20 +652,153 @@ func searchPkgViaHasSBOM(ctx context.Context, gqlclient graphql.Client, searchSt
 	for _, result := range collectedPkgVersionResults {
 		for _, neighbor := range result.pkgVersionNeighborResponse.Neighbors {
 			if certifyVuln, ok := neighbor.(*model.NeighborsNeighborsCertifyVuln); ok {
-				if !checkedCertifyVulnIDs[certifyVuln.Vulnerability.Id] {
-					if certifyVuln.Vulnerability.Type != noVulnType {
-						checkedCertifyVulnIDs[certifyVuln.Vulnerability.Id] = true
-						for _, vuln := range certifyVuln.Vulnerability.VulnerabilityIDs {
-							tableRows = append(tableRows, table.Row{certifyVulnStr, certifyVuln.Id, "vulnerability ID: " + vuln.VulnerabilityID})
-							path = append(path, []string{vuln.Id, certifyVuln.Id,
-								certifyVuln.Package.Namespaces[0].Names[0].Versions[0].Id,
-								certifyVuln.Package.Namespaces[0].Names[0].Id, certifyVuln.Package.Namespaces[0].Id,
-								certifyVuln.Package.Id}...)
+				if !checkedCertifyVulnIDs[certifyVuln.Id] && certifyVuln.Vulnerability.Type != noVulnType {
+					checkedCertifyVulnIDs[certifyVuln.Id] = true
+					for _, vuln := range certifyVuln.Vulnerability.VulnerabilityIDs {
+						tableRows = append(tableRows, table.Row{certifyVulnStr, certifyVuln.Id, "vulnerability ID: " + vuln.VulnerabilityID})
+						path = append(path, []string{vuln.Id, certifyVuln.Id,
+							certifyVuln.Package.Namespaces[0].Names[0].Versions[0].Id,
+							certifyVuln.Package.Namespaces[0].Names[0].Id, certifyVuln.Package.Namespaces[0].Id,
+							certifyVuln.Package.Id}...)
+					}
+					path = append(path, result.isDep.Id, result.isDep.Package.Namespaces[0].Names[0].Versions[0].Id,
+						result.isDep.Package.Namespaces[0].Names[0].Id, result.isDep.Package.Namespaces[0].Id,
+						result.isDep.Package.Id)
+				}
+			}
+
+			if certifyVex, ok := neighbor.(*model.NeighborsNeighborsCertifyVEXStatement); ok {
+				for _, vuln := range certifyVex.Vulnerability.VulnerabilityIDs {
+					tableRows = append(tableRows, table.Row{vexLinkStr, certifyVex.Id, "vulnerability ID: " + vuln.VulnerabilityID + ", Vex Status: " + string(certifyVex.Status) + ", Subject: " + vexSubjectString(certifyVex.Subject)})
+					path = append(path, certifyVex.Id, vuln.Id)
+				}
+				path = append(path, vexSubjectIds(certifyVex.Subject)...)
+			}
+		}
+	}
+	return path, tableRows, nil
+}
+
+func searchForSBOMViaArtifact(ctx context.Context, gqlclient graphql.Client, searchString string, maxLength int, primaryCall bool) ([]string, []table.Row, error) {
```

Comment:
I think we should have the search functions in a central place so that we don't duplicate too many DFS / BFSes with a lot of boilerplate code - in addition we want to use the same type of search (in this case by SBOM) for everything, and having a common function will be helpful! Can you help take a look at https://github.com/guacsec/guac/blob/0675b6779487e206f30d64ce234bb42c762502a3/pkg/guacanalytics/patchPlanning.go and see if we can implement it in that function?

Reply:
Good idea, I have moved the search functionalities to a separate file. Thank you!